Library/Homebrew/tap.rb

```diff
@@ -280,4 +305,70 @@ def self.each
   def self.names
     map(&:name)
   end
+
+  private
+
+  # @private
+  def formula_file_to_name(file)
+    "#{name}/#{file.basename(".rb")}"
+  end
+
+  # @private
+  def alias_file_to_name(file)
+    "#{name}/#{file.basename}"
+  end
+end
+
+class CoreTap < Tap
```

Comment:
It will be more like tap in the future :wink: And they do share lots of similarities.